prow/gerrit/adapter/adapter.go

```diff
@@ -91,38 +91,54 @@ func NewController(ctx context.Context, prowJobClient prowv1.ProwJobInterface, o
 	if err != nil {
 		logrus.WithError(err).Fatal("Error creating gerrit client.")
 	}
-	gerritClient.Authenticate(cookiefilePath, tokenPathOverride)
-
 	c := &Controller{
 		prowJobClient:      prowJobClient,
 		config:             cfg,
 		gc:                 gerritClient,
 		tracker:            lastSyncTracker,
 		projectsOptOutHelp: projectsOptOutHelpMap,
 	}
+
+	// applyGlobalConfig reads gerrit configurations from global gerrit config,
+	// it will completely override previously configured gerrit hosts and projects.
+	// it will also by the way authenticate gerrit
+	c.applyGlobalConfig(cfg, gerritClient, lastSyncTracker, cookiefilePath, tokenPathOverride)
+
+	return c
+}
+
+func (c *Controller) applyGlobalConfig(cfg config.Getter, gerritClient *client.Client, lastSyncTracker *syncTime, cookiefilePath, tokenPathOverride string) {
+	c.applyGlobalConfigOnce(cfg, gerritClient, lastSyncTracker, cookiefilePath, tokenPathOverride)
+
 	go func() {
 		for {
-			orgReposConfig := cfg().Gerrit.OrgReposConfig
-			if orgReposConfig == nil {
-				time.Sleep(time.Second)
```

Comment:
Rather than checking every second and always processing if any config exists we can subscribe to config change notifications from the config agent: https://github.com/kubernetes/test-infra/blob/b5a1578a0c653b9d68a3c78ff7bd9b38f87a063f/prow/config/agent.go#L376-L384

Reply:
I'll take a look, the usage of this function is really low though